packages/string-templates/src/helpers/javascript.js

```diff
@@ -48,14 +49,26 @@ module.exports.processJS = (handlebars, context) => {
   try {
     // Wrap JS in a function and immediately invoke it.
     // This is required to allow the final `return` statement to be valid.
-    const js = `(function(){${atob(handlebars)}})();`
+    const js = iifeWrapper(atob(handlebars))
 
     // Our $ context function gets a value from context.
     // We clone the context to avoid mutation in the binding affecting real
     // app context.
     const sandboxContext = {
       $: path => getContextValue(path, cloneDeep(context)),
       helpers: getJsHelperList(),
+
+      // Proxy to evaluate snippets when running in the browser
+      snippets: new Proxy(
+        {},
+        {
+          get: function (_, name) {
+            // This will error if the snippet doesn't exist, but that's intended
+            const snippet = (context.snippets || []).find(x => x.name === name)
```

Comment:
Is it worth gathering these snippets into an object so that accessing them doesn't require a scan across all snippets? I don't imagine it's going to make the world of difference, but potentially unbounded code makes me nervous.

Reply:
Every little helps! ™️